operator/v1/types_network.go

```diff
@@ -100,12 +100,90 @@ type DefaultNetworkDefinition struct {
 	KuryrConfig *KuryrConfig `json:"kuryrConfig,omitempty"`
 }
 
+// SimpleMacvlanConfig contains configurations for macvlan interface.
+type SimpleMacvlanConfig struct {
+	// master is the host interface to create the macvlan interface from.
+	// If not specified, it will be default route interface
+	// +optional
+	Master string `json:"master,omitempty"`
+
+	// IPAMConfig configures IPAM module will be used for IP Address Management (IPAM).
+	// +optional
+	IPAMConfig *IPAMConfig `json:"ipamConfig,omitempty"`
+
+	// mode is the macvlan mode: bridge, private, vepa, passthru. The default is bridge
+	// +optional
+	Mode MacvlanMode `json:"mode,omitempty"`
+
+	// mtu is the mtu to use for the macvlan interface. if unset, host's
+	// kernel will select the value.
+	// +optional
+	MTU uint32 `json:"mtu,omitempty"`
+}
+
+// StaticIPAMAddresses provides IP address and Gateway for static IPAM addresses
+type StaticIPAMAddresses struct {
+	// Address is the IP address in CIDR format
+	// +optional
+	Address string `json:"address"`
+	// Gateway is IP inside of subnet to designate as the gateway
+	// +optional
+	Gateway string `json:"gateway,omitempty"`
+}
+
+// StaticIPAMRoutes provides Destination/Gateway pairs for static IPAM routes
+type StaticIPAMRoutes struct {
+	// Destination points the IP route destination
+	Destination string `json:"destination"`
+	// Gateway is the route's next-hop IP address
+	// If unset, a default gateway is assumed (as determined by the CNI plugin).
+	// +optional
+	Gateway string `json:"gateway,omitempty"`
+}
+
+// StaticIPAMDNS provides DNS related information for static IPAM
+type StaticIPAMDNS struct {
+	// Nameservers points DNS servers for IP lookup
+	// +optional
+	Nameservers []string `json:"nameservers,omitempty"`
+	// Domain configures the domainname the local domain used for short hostname lookups
+	// +optional
+	Domain string `json:"domain,omitempty"`
+	// Search configures priority ordered search domains for short hostname lookups
+	// +optional
+	Search []string `json:"search,omitempty"`
+}
+
+// StaticIPAMConfig contains configurations for static IPAM (IP Address Management)
+type StaticIPAMConfig struct {
+	// Addresses configures IP address for the interface
+	// +optional
+	Addresses []StaticIPAMAddresses `json:"addresses,omitempty"`
```

Comment:
Is there a use case for this? Do we expect to set up networks where the address is completely hard-coded?  Presumably, then, that network will have one pod.

If so, that's fine. But if not, then is there a use to putting this in the config?

Reply:
You're right.